### ISSUE TITLE
Mismatch in resnet50 error rates for pretrained model.

### ISSUE THREAD

**Issue by idharmateja** [pytorch/examples#389]
In https://pytorch.org/docs/stable/torchvision/models.html , reported top-1 and top-5 errors for pretrained resnet50 model are 23.85 and 7.13 respectively. But when I run on my system, I get 24.30 and 7.35 respectively.  What am I missing here ?

**command**
` python examples/imagenet/main.py -a resnet50 --pretrained -e  /scratch/dvooturi/datasets/imagenet/`

**stdout**
=> using pre-trained model 'resnet50'
Test: [0/196]   Time 4.590 (4.590)      Loss 0.5068 (0.5068)    Prec@1 85.156 (85.156)  Prec@5 97.656 (97.656)
Test: [10/196]  Time 0.350 (0.712)      Loss 0.9695 (0.6843)    Prec@1 77.734 (82.102)  Prec@5 93.359 (95.561)
Test: [20/196]  Time 1.489 (0.658)      Loss 0.8174 (0.7009)    Prec@1 85.547 (81.808)  Prec@5 91.406 (95.182)
Test: [30/196]  Time 0.233 (0.603)      Loss 0.7984 (0.6579)    Prec@1 81.250 (83.165)  Prec@5 95.312 (95.602)
Test: [40/196]  Time 1.421 (0.602)      Loss 0.6284 (0.7015)    Prec@1 83.203 (81.641)  Prec@5 97.266 (95.589)
Test: [50/196]  Time 0.233 (0.579)      Loss 0.4582 (0.6999)    Prec@1 89.453 (81.549)  Prec@5 97.266 (95.695)
Test: [60/196]  Time 1.337 (0.584)      Loss 0.9202 (0.7121)    Prec@1 76.562 (81.231)  Prec@5 95.312 (95.748)
Test: [70/196]  Time 0.233 (0.573)      Loss 0.7442 (0.6995)    Prec@1 78.516 (81.553)  Prec@5 96.484 (95.896)
Test: [80/196]  Time 1.503 (0.579)      Loss 1.5054 (0.7265)    Prec@1 61.328 (81.028)  Prec@5 87.500 (95.554)
Test: [90/196]  Time 0.233 (0.569)      Loss 1.8334 (0.7735)    Prec@1 54.688 (80.001)  Prec@5 86.719 (95.025)
Test: [100/196] Time 1.210 (0.571)      Loss 1.1607 (0.8253)    Prec@1 66.406 (78.833)  Prec@5 91.797 (94.462)
Test: [110/196] Time 0.233 (0.565)      Loss 0.8393 (0.8496)    Prec@1 76.562 (78.273)  Prec@5 93.750 (94.197)
Test: [120/196] Time 1.093 (0.567)      Loss 1.3103 (0.8676)    Prec@1 67.578 (77.970)  Prec@5 86.328 (93.886)
Test: [130/196] Time 0.232 (0.564)      Loss 0.6996 (0.8999)    Prec@1 82.031 (77.144)  Prec@5 95.312 (93.562)
Test: [140/196] Time 1.077 (0.563)      Loss 1.0345 (0.9172)    Prec@1 76.172 (76.870)  Prec@5 92.578 (93.368)
Test: [150/196] Time 0.235 (0.561)      Loss 1.0517 (0.9354)    Prec@1 75.391 (76.573)  Prec@5 90.625 (93.072)
Test: [160/196] Time 1.270 (0.561)      Loss 0.7198 (0.9496)    Prec@1 85.156 (76.303)  Prec@5 94.531 (92.867)
Test: [170/196] Time 0.233 (0.558)      Loss 0.6221 (0.9661)    Prec@1 83.984 (75.907)  Prec@5 98.047 (92.692)
Test: [180/196] Time 1.077 (0.558)      Loss 1.2556 (0.9814)    Prec@1 68.750 (75.591)  Prec@5 91.797 (92.563)
Test: [190/196] Time 0.232 (0.557)      Loss 1.1806 (0.9805)    Prec@1 68.750 (75.587)  Prec@5 96.875 (92.613)
 * Prec@1 75.704 Prec@5 92.654

**Comment by soumith** [comment]
I verified this quite recently, so I'm surprised by your result. What version of PyTorch and torchvision are you using? and on what OS?

**Comment by idharmateja** [comment]
I reran it again and got the same results ! Following is the config:
torch = 0.4.0
torchvision = 0.2.1
OS = CentOS Linux release 7.4.1708 (Core)

I am using code at commit -> 98b1fce0d2e55c79b59e98ec28ebbc2f58da38f7 .

**Comment by soumith** [comment]
I just re-ran now with torch 0.4.0 and torchvision 0.2.1.
My result is: ` * Prec@1 76.130 Prec@5 92.862`

Did you resize / downsample the validation images before-hand from their original resolution?

**Comment by idharmateja** [comment]
I was using existing dataset path on our cluster. To isolate the problem, I prepared val folder from scratch using tar file downloaded from imagenet site. Now I get the same result as yours.

The run which produced same results was in python 3.6 environment. But the same run with python 2.7 environment and same versions or torch and torchvision gives  `Prec@1 76.012 Prec@5 92.934` . 

Anyways, thanks for the quick response @soumith .

**Comment by soumith** [comment]
> But the same run with python 2.7 environment and same versions or torch and torchvision gives Prec@1 76.012 Prec@5 92.934 .

Because of CuDNN, the runs might be a little non-deterministic.
if you want determinism at the expense of speed, you can set:
```
torch.backends.cudnn.deterministic=True
torch.backends.cudnn.benchmark=False
```
